### ISSUE TITLE
Godot 4.0 alpha2 AnimatedSprite3d "p_elem->_root != this" error

### ISSUE THREAD

**Issue by cptfurball** [godotengine/godot#58348]
### Godot version

v4.0.alpha2.official [79077e6c1]

### System information

Windows 10, Vulkan, GTX1660 Super

### Issue description

When switching animation via GDScript in Godot Alpha, there are errors that occur every single tick (I assume). Error message is as follow:

```
E 0:00:03:0596   remove: Condition "p_elem->_root != this" is true.
  <C++ Source>   ./core/templates/self_list.h:80 @ remove()
```

This only happens when the animation is playing. It seems to have some correlation with frames as the errors occurs per animation frame. Reducing the animation fps slows down the error rate. That is all I could provide with my limited knowledge.

This is the first time I am reporting an issue. I hope I did this right for the community. Please let me know if I needed more clarification. Sample project is provided in the zip file.

### Steps to reproduce

```
extends Node3D


func _on_timer_timeout():
	if $AnimatedSprite3D.get_animation() == "default":
		$AnimatedSprite3D.play("another_anim")
	else:
		$AnimatedSprite3D.play("default")
```

This is the basic code to reproduce the error.

The AnimatedSprite3d must have "another_anim" and "default" animation created. It does not matter if it has more than one frames or just one frame.

### Minimal reproduction project

[animated-sprite-error.zip](https://github.com/godotengine/godot/files/8103685/animated-sprite-error.zip)

**Comment by timothyqiu** [comment]
Seems fixed. I can reproduce this with `alpha2`, but not current `master`.

**Comment by cptfurball** [comment]
Alright, let me try again with the latest master. Thanks for confirming. I shouldve checked the latest before reporting this.

**Comment by akien-mga** [comment]
Duplicate of https://github.com/godotengine/godot/issues/55682

**Comment by cptfurball** [comment]
Confirmed fixed. Thanks :D
